### DISCUSSION TITLE
Sessão  para exibir novas faq adicionadas / as mais procuradas

### DISCUSSION THREAD

**Discussion by Morpa** [Morpa/FAQ#18] [Ideas]
@GuilleAngulo acho que podemos até fazer um refactor mais profundo. Quando comecei a fazer esse projeto estavamos ainda no início do curso  e de lá até agora aprendemos muito mais coisas.
Não sei se ainda é válido usar o MD como fonte de dados ou criar um backend no strapi e uma página de formulário público (onde posteriormente validariamos no painel admin).
Se você tiver alguma idéia a mais fique a vontade. Te enviei um invite para ser colaborador.

Vamos discutindo por aqui.

= D

**Comment by GuilleAngulo** [comment]
Opa, valeu pelo convite! 

Acho que para desenvolver essas funcionalidades sim vale a pena ter algum backend, pode até funcionar algum pequeno tipo mongodb em cloud pensando que o conteúdo armazenado seria só basicamente os textos talvez vale a pena pensar em uma estrutura menos complexa que o strapi? o que você acha?

Em algum momento pensei que outra opção podia ser criar algum processo automatizado que chama-se directamente o README do repo com as FAQs. Dessa maneira o ponto "oficial" de abrir novas seria lá. Mas acho maneiro criar um formulário para criar novas também, me ocorre que ai poderíamos para começar adicionar um estado "draft" a todas as novas e uma vez validadas (no banco mesmo) mudar para um "approved" por exemplo (também no banco mesmo para começar). Aberto para qualquer abordagem que você decida ✌

**Reply by Morpa** [reply]
É nesse sentido você tem razão. O strapi acabaria sendo muito complexo para uma coisa tão simples. 
Eu até criei algumas API's no passado, mas estou meio enferrujado. Não sei se você está mais confortável para fazer isso.
Aí teríamos que criar também um painel administrativo para a aprovação das FAQs.
Se quiser esboçar alguma coisa de backend segue em frente. Durante essa semana vou pensar aqui em alguma coisa.

**Reply by willianjusten** [reply]
Que tal usar um serviço simples, do tipo um GraphCMS e só expor a mutation num formulário? Ficaria bem fácil de fazer e num teria que se preocupar com backend, onde servir e blablabla

**Reply by Morpa** [reply]
Essa foi uma bela idéia 🙌🏻.  Vou ver se essa semana encontro um tempinho para ver isso!

**Reply by GuilleAngulo** [reply]
Ótimo conselho! E uma boa oportunidade para conhecer o GraphCMS. Vou pesquisar também essa semana na documentação para conhecer mais sobre

**Comment by GuilleAngulo** [comment]
Antes de ir para o GraphCMS podemos definir que campos vaõ ter as entidades FAQ para ter uma ideia (sem precissar ser definitivos). Aqui vai uma sugestão para discutir:

```
title: string
slug: string
content: string /* markdown */
published: boolean /* false = rascunho | true = aprobado */
reference: {  /* Para facilitar as buscas por modulo, seção ou aula. São necessarios os tres? O modulo o unico obrigatorio? */
  module: number, 
  section?: number,
  lesson?: number
} 
created_at: number
```
Quais campos voce acha que deveriam ser adicionados? Para as faqs mais populares a abordagem seria com um contador de acessos da pagina ou melhor com "estrelas/likes" que o usuário da se a pergunta em questão for útil?

**Reply by Morpa** [reply]
A princípio achei interessante essa estrutura. Podemos começar por ela. Acho que os likes retratariam mais a realidade, pois a pessoa pode acessar e nem ler ou ter se enganado. Ai tipo colocariamos "Essa dica te ajudou?" (Não sei se um sim ou não), mas caso a pessoa clicar no sim por exemplo adicionaria um ponto a mais para ela na base. 

Ficaria meio que assim:
```js
title: string
slug: string
content: string /* markdown */
published: boolean /* false = rascunho | true = aprobado */
reference: {  /* Para facilitar as buscas por modulo, seção ou aula. São necessarios os tres? O modulo o unico obrigatorio? */
  module: number, 
  section?: number,
  lesson?: number
} 
points: number
created_at: number `string or date?`
```
